integration_tests/tf_distribute_training_test.py

```diff
@@ -49,6 +49,7 @@ def test_model_fit():
             ),
             loss=losses.MeanSquaredError(),
             metrics=[metrics.MeanSquaredError()],
+            jit_compile=True,
```

Comment:
This test is now failing with a device related issue:

```
Trying to access resource dense/kernel/replica_1/3 (defined @ /home/runner/work/keras/keras/keras/backend/tensorflow/core.py:31) located in device /job:localhost/replica:0/task:0/device:CPU:1 from device /job:localhost/replica:0/task:0/device:CPU:0
 Cf. https://www.tensorflow.org/xla/known_issues#tfvariable_on_a_different_device
```

I believe this issue is what would prevent us from enabling XLA with tf.distribute (rather than the variable creation issue). Maybe the issue is general to XLA + tf.distribute, or maybe it is specific to how the current optimizers are implemented.

There was this related issue that was filed previously for tf.keras: https://github.com/keras-team/tf-keras/issues/385

Reply:
Thanks for the reference of the issue, let me take a closer look. 

In general, XLA should work with tf.distribution. Basically it will try to jit compile the train function for each of the device. I think we are probably missing something in the new optimizer implementation (comparing to legacy optimizer).

Reply:
Humm, I take this back. It seems that there is a long standing issue for XLA + distribution since https://github.com/tensorflow/tensorflow/issues/53140, and it seems to be a distribution issue for device placement. For now, let's disable the XLA when using distribution.